File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.js
function solution(arr) {
  const n = arr.length;
  if(!(n & (n - 1))) return arr
  else{
    for (let i = 1; i <= 512; i++) {
      let d = !(n + i & (n + i - 1))
      arr.push(0 * i)
      if(d === true) break
    }
  }
  return arr
}

File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] arr) {
        int length1 = arr.length;
        int length2 = 1;
        
        while (length2 < length1) {
            length2 *= 2;
        }
        
        if(length2 == length1) {
            return arr;
        }
        
        int[] answer = Arrays.copyOf(arr, length2);
        return answer;
    }
}